src/lib/components/SelectField/SelectField.jsx

```diff
@@ -69,16 +70,20 @@ export const SelectField = ({
             value={value}
           >
             {
-              options.map((option) => (
-                <option
-                  disabled={option.disabled}
-                  id={id && `${id}__item__${option.value}`}
-                  key={option.value}
-                  value={option.value}
-                >
-                  {option.label}
-                </option>
-              ))
+              options.map((option) => {
+                if ('options' in option) {
+                  return (
+                    <optgroup label={option.label} key={option.label}>
```

Comment:
Here `label` as is used as `key`. In the `Radio` component, we use `value` as `key`.

I'm not sure what is better:
* `label` can be translated and can contain strange characters. On the other hand, two `<options>` or `<optgroups>` with same `label` make no sense. It can probably happen in case of a missing translation or something like that.
* `value` can be same on two `<options>`. It is strange, but i can imagine this happening more ofthen then the same `label`
* we could use `nanoid`, but that is not elegant

Reply:
see: https://github.com/react-ui-org/react-ui/pull/401/commits/54c7af01f3b60452b67f71ba5c5b4e96a9e1afad#r943341569